### PR TITLE
Feature/spinedem 2176 allocate nhs number tests

### PR DESCRIPTION
**Pull request by pl-nhs**
## Summary
* Routine Change

This is a functional test for the nhs number allocation work

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.

### CODE REVIEW

File: tests/functional/features/post_patient.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    When method POST
+    Then status 201
+    And expected_response template == read(valid_patient_post_response)
+    And set expected_response['#nhsNumber'] = response['id']
+    And set expected_response['#family'] = 'Smith'
+    And ignore in response comparison the family name Id 
+    And response body == expected_response
+
+
+@ignore
```
Comment (davidhamill1-nhs) [inline tests/functional/features/post_patient.feature:64]: Is there a reason this is (presumably) ignored?

File: tests/functional/test_patient_create.py
Comment (davidhamill1-nhs) [inline tests/functional/test_patient_create.py:1]: I don't think this file gets run in any pipeline. It'll needing adding to a `azure/templates/pds-tests.yml`.

File: azure/templates/pds-tests.yml [hunk 2]
```diff
@@ -120,3 +120,16 @@ steps:
...
+      export APIGEE_API_TOKEN="$(secret.AccessToken)"
+      export APIGEE_ACCESS_TOKEN="$(secret.AccessToken)"
+      export OAUTH_BASE_URI="https://$(ENVIRONMENT).api.service.nhs.uk"
+      export OAUTH_PROXY="oauth2-mock"
+      export TEST_PATIENT_ID="$(TEST_PATIENT_ID)"
+      export INTERNAL_DEV_ASID="$(INTERNAL_DEV_ASID)"
+
+      poetry run pytest -v tests/functional/test_patient_create.py --api-name=personal-demographics-service --proxy-name "$(FULLY_QUALIFIED_SERVICE_NAME)" --junitxml=tests/patient-access-test-report.xml --reruns 20 --reruns-delay 1
+    displayName: Run post patient (create NHS number) tests
+    workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)"
```
Comment (davidhamill1-nhs) [inline azure/templates/pds-tests.yml:135]: This is not consistently applied to all test files run, but probably worth taking advantage of publishing tests.
```suggestion
    workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)"
    
  - task: PublishTestResults@2
    displayName: "Publish sync wrap test results"
    inputs:
      testResultsFiles: $(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)/tests/patient-create-test-report.xml
      failTaskOnFailedTests: true
```

File: azure/templates/pds-tests.yml [hunk 3]
```diff
@@ -120,3 +120,16 @@ steps:
...
+
+  - bash: |
+      export APIGEE_API_TOKEN="$(secret.AccessToken)"
+      export APIGEE_ACCESS_TOKEN="$(secret.AccessToken)"
+      export OAUTH_BASE_URI="https://$(ENVIRONMENT).api.service.nhs.uk"
+      export OAUTH_PROXY="oauth2-mock"
+      export TEST_PATIENT_ID="$(TEST_PATIENT_ID)"
+      export INTERNAL_DEV_ASID="$(INTERNAL_DEV_ASID)"
+
+      poetry run pytest -v tests/functional/test_patient_create.py --api-name=personal-demographics-service --proxy-name "$(FULLY_QUALIFIED_SERVICE_NAME)" --junitxml=tests/patient-access-test-report.xml --reruns 20 --reruns-delay 1
```
Comment (davidhamill1-nhs) [inline azure/templates/pds-tests.yml:133]: Change report file name.
```suggestion
      poetry run pytest -v tests/functional/test_patient_create.py --api-name=personal-demographics-service --proxy-name "$(FULLY_QUALIFIED_SERVICE_NAME)" --junitxml=tests/patient-create-test-report.xml --reruns 20 --reruns-delay 1
```

File: azure/templates/pds-tests.yml [hunk 4]
```diff
@@ -120,3 +120,16 @@ steps:
       poetry run pytest -v tests/functional/test_patient_access.py --api-name=personal-demographics-service --proxy-name "$(FULLY_QUALIFIED_SERVICE_NAME)" --junitxml=tests/patient-access-test-report.xml --reruns 20 --reruns-delay 1
     displayName: Run patient access tests
     workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)"
```
Comment (davidhamill1-nhs) [inline azure/templates/pds-tests.yml:122]: This hasn't had its test results published.
```suggestion
    workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)"
    
  - task: PublishTestResults@2
    displayName: "Publish sync wrap test results"
    inputs:
      testResultsFiles: $(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)/tests/patient-access-test-report.xml
      failTaskOnFailedTests: true
```

**Review by davidhamill1-nhs — CHANGES_REQUESTED** [review]

File: tests/functional/test_patient_create.py
Comment (davidhamill1-nhs) [reply tests/functional/test_patient_create.py:1]: You've now done this. Thanks

File: tests/functional/features/post_patient.feature (hunk 1 above)
Comment (pl-nhs) [reply tests/functional/features/post_patient.feature:64]: Nope, I added that tag and then left it by accident...

**Review by davidhamill1-nhs — APPROVED** [review]

**Pull request merged**